lib/src/utf8.dart

```diff
@@ -8,54 +8,93 @@ import 'dart:typed_data';
+  }
+
+  /// Converts the UTF-8 encoded [string] to a Dart string.
+  ///
+  /// Decodes the UTF-8 code units of this zero-terminated byte array as
+  /// Unicode code points and creates a Dart string containing those code
+  /// points.
+  ///
+  /// If [length] is provided, zero-termination is ignored and the result can
+  /// contain NUL characters.
+  @Deprecated('Use Utf8Pointer.toDartString instead.')
+  static String fromUtf8(Pointer<Utf8> string, {int? length}) {
+    return string.toDartString(length: length);
+  }
+
+  /// Creates a zero-terminated [Utf8] code-unit array from [string].
+  ///
+  /// If [string] contains NUL characters, the converted string will be truncated
+  /// prematurely. Unpaired surrogate code points in [string] will be encoded
+  /// as replacement characters (U+FFFD, encoded as the bytes 0xEF 0xBF 0xBD)
+  /// in the UTF-8 encoded result. See [Utf8Encoder] for details on encoding.
+  ///
+  /// Returns an [allocator]-allocated pointer to the result.
+  @Deprecated('Use StringUtf8Pointer.toNativeUtf8 instead.')
+  static Pointer<Utf8> toUtf8(String string, {Allocator allocator = calloc}) {
+    return string.toNativeUtf8(allocator: allocator);
+  }
+}
+
+extension Utf8Pointer on Pointer<Utf8> {
+  /// The number of UTF-8 code units in this zero-terminated UTF-8 string.
+  ///
+  /// The UTF-8 code units of the strings are the non-zero code units up to the
+  /// first zero code unit.
+  int get length {
+    final Pointer<Uint8> array = cast<Uint8>();
     int length = 0;
     while (array[length] != 0) {
       length++;
     }
     return length;
   }
 
-  /// Creates a [String] containing the characters UTF-8 encoded in [string].
+  /// Converts this UTF-8 encoded string to a Dart string.
   ///
-  /// Either the [string] must be zero-terminated or its [length] &mdash; the
-  /// number of bytes &mdash; must be specified as a non-negative value. The
-  /// byte sequence must be valid UTF-8 encodings of Unicode scalar values. A
-  /// [FormatException] is thrown if the input is malformed. See [Utf8Decoder]
-  /// for details on decoding.
+  /// Decodes the UTF-8 code units of this zero-terminated byte array as
+  /// Unicode code points and creates a Dart string containing those code
+  /// points.
   ///
-  /// Returns a Dart string containing the decoded code points.
-  static String fromUtf8(Pointer<Utf8> string, {int? length}) {
+  /// If [length] is provided, zero-termination is ignored and the result can
+  /// contain NUL characters.
+  String toDartString({int? length}) {
     if (length != null) {
       RangeError.checkNotNegative(length, 'length');
     } else {
-      length = strlen(string);
+      length = this.length;
     }
-    return utf8.decode(string.cast<Uint8>().asTypedList(length));
+    return utf8.decode(cast<Uint8>().asTypedList(length));
   }
+}
 
-  /// Convert a [String] to a UTF-8 encoded zero-terminated C string.
+extension StringUtf8Pointer on String {
```

Comment:
Maybe combine the `StringUtf8Pointer` and `StringUtf16Pointer` extensions into one, maybe called `StringUtfPointer`?
I can't see any situation where a user would want to import one and hide the other.

Your call.

Reply:
True. But I'd like to keep them in separate files.